SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -1154,6 +1154,9 @@ - (void)handleSendToMyself:(TSOutgoingMessage *)outgoingMessage
         }
 
         // We want the incoming message to appear after the outgoing message.
+        //
+        // TODO: We need to be careful to duplicate the attachments for
```

Comment:
Sorry this wasn't more clear.  

Note that this code was in `handleSendToMyself`.

Essentially we want to avoid re-introducing a bug we've already fixed for conventional "body" attachments with the new attachments for quoted replies and shared contact avatars.  

The bug is: We send a "contact share" (for example) to _ourselves_ in a 1:1 thread with ourself.  We create two instances of TSMessage: one incoming, one outgoing.  We then delete one of those two messages but not the other.  The other should remain valid.  

The naive way to have an incoming/outgoing message pair in a "self" thread is to have them share the attachment, but that will break when one is deleted.


There's already a card.